x/sync/client.go

```diff
@@ -31,177 +25,25 @@ const (
 )
 
 var (
-	_ Client = (*client)(nil)
-
 	errInvalidRangeProof             = errors.New("failed to verify range proof")
 	errInvalidChangeProof            = errors.New("failed to verify change proof")
 	errTooManyKeys                   = errors.New("response contains more than requested keys")
 	errTooManyBytes                  = errors.New("response contains more than requested bytes")
 	errUnexpectedChangeProofResponse = errors.New("unexpected response type")
 )
 
-// Client synchronously fetches data from the network
-// to fulfill state sync requests.
-// Repeatedly retries failed requests until the context is canceled.
 type Client interface {
```

Comment:
Pretty annoying - we need to have this interface still for testing purposes. This is because we need to be able to mock timeouts to test our retries because the p2p package does not actually send app errors yet. This can be removed once https://github.com/ava-labs/avalanchego/pull/2753 is merged.